src/workerd/api/actor.h

```diff
@@ -13,20 +13,62 @@
 #include <capnp/compat/http-over-capnp.h>
 #include <workerd/api/http.h>
 #include <workerd/jsg/jsg.h>
+#include <workerd/jsg/function.h>
 
 namespace workerd {
   template <typename T> class IoOwn;
 }
 
 namespace workerd::api {
 
+// This class is the client RPC stub for Durable Objects. It intercepts dynamic method calls, then
+// sends the method name and arguments to the remote Durable Object, returning the response it
+// receives.
+//
+// We don't want name interception on Fetcher itself -- at least not yet -- since not all Fetcher
+// implementations need it. Name interception is off by default; to turn it on, set the JsRpcStub
+// compatibility flag.
+class DurableObjectRpcStub : public Fetcher, public jsg::NamedIntercept {
+public:
+  DurableObjectRpcStub(
+      IoOwn<OutgoingFactory> outgoingFactory,
+      RequiresHostAndProtocol requiresHost,
+      bool inHouse)
+    : Fetcher(kj::mv(outgoingFactory), requiresHost, inHouse) {}
+
+  kj::Maybe<jsg::JsValue> getNamed(jsg::Lock& js, kj::StringPtr name) override {
```

Comment:
This method is probably too complicated to inline, so should be defined in the `.c++` file, not the header.

(This is especially true after the next PR that actually implements the RPC here.)

Reply:
As part of the code movement in the other comment, I moved the implementation of `getNamed()` into the c++ file.